plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryTypeUtils.java

```diff
@@ -87,6 +95,12 @@ public static Object readNativeValue(Type type, Block block, int position)
             long days = type.getLong(block, position);
             return DATE_FORMATTER.format(LocalDate.ofEpochDay(days));
         }
+        if (type.equals(TIMESTAMP_MICROS)) {
```

Comment:
I probably miss it but where is the code which maps BigQuery datetime to TIMESTAMP_MICROS?

Reply:
Oh I see it isn't needed. The BigQueryPageSource returns blocks of proper type because BigQueryColumnHandle stores the Trino type for give column which page source uses to produce proper blocks.